src/EditorComponent.jsx

```diff
@@ -235,18 +234,28 @@ export default class EditorComponent extends Component {
 
         this.editorRef = undefined;
         this.hoverMenuRef = undefined;
+        this._makePlugins();
+        this.selectionBeforeBlur = undefined;
     }
 
-    onChange = ({ value }) => {
-        if (value.document !== this.state.value.document) {
-            this.saveCurrentNote();
+    onChange = change => {
+        let { value } = change;
+
+        // ^ The value that onChange receives as argument is the new value of the editor.
+        // Main reason we are overriding is to setState with the new value.
+        if (AUTOSAVE && value.document !== this.state.value.document) {
+            this.saveCurrentNote(new SaveCurrentNoteOptions(false, 0, false, true));
         }
 
-        // Check if we are on the boundary of a timestamp mark. If so we
-        // will toggle away that mark state.
-        const onTimestamp = value.marks.some(({ type }) => type === 'youtube_timestamp');
+        // We can call mathjax to typeset the page here. TODO(rksht): don't tell it to update only
+        // when there's at least one inline math element in the block that is currently being
+        // edited?
+        // MathJax.Hub.Queue(['Typeset', MathJax.Hub, '__editor_container_div__']);
 
-        this.setState({ onTimestamp, value });
+        // Check if we are on the boundary of a timestamp mark. If so we will toggle away that mark state.
+        const onTimestamp = value.marks.some(mark => mark.type === 'youtube_timestamp');
+        this.setState({ onTimestamp });
+        this.setState({ value });
```

Comment:
Could be `this.setState({ onTimestamp, value })`. Multiple setState calls add complexity.